Float.TinCan.LocalLRSServer/LRSServer.cs

```diff
@@ -247,66 +242,85 @@ void HandleStateRequest(HttpListenerRequest request, HttpListenerResponse respon
-
-            if (method == HttpMethod.Get)
+            try
             {
-                if (string.IsNullOrWhiteSpace(profileIdString))
+                switch (method)
                 {
-                    // todo: return all available IDs
+                    case string m when HttpMethod.Get.ToString() == method:
+                        if (string.IsNullOrWhiteSpace(profileIdString))
+                        {
+                            // todo: return all available IDs
 #pragma warning disable CA1303 // Do not pass literals as localized parameters
-                    throw new NotImplementedException("GET requests for all documents are not yet implemented");
+                            throw new NotImplementedException("GET requests for all documents are not yet implemented");
 #pragma warning restore CA1303 // Do not pass literals as localized parameters
-                }
-
-                if (serverDelegate != null)
-                {
-                    profileDocument = serverDelegate.AgentProfileDocumentForProfileId(profileIdString);
-                }
-            }
-            else if (method == HttpMethod.Delete)
-            {
-                // todo: implement delete requests
+                        }
+
+                        if (serverDelegate != null)
+                        {
+                            profileDocument = serverDelegate.AgentProfileDocumentForProfileId(profileIdString);
+                        }
+
+                        WriteToStream(response, profileDocument.content, ContentType.Json, HttpStatusCode.OK);
+                        return;
+                    case string m when HttpMethod.Post.ToString() == method:
+                    case string m1 when HttpMethod.Put.ToString() == method:
+                        profileDocument = new AgentProfileDocument
+                        {
+                            id = WebUtility.UrlDecode(profileIdString),
+                        };
+
+                        var agentString = WebUtility.UrlDecode(request.QueryString["agent"]);
+
+                        if (agentString != null)
+                        {
+                            profileDocument.agent = new Agent(new StringOfJSON(agentString));
+                        }
+
+                        Encoding encoding = request.ContentEncoding;
+                        var bytes = new byte[request.ContentLength64];
+                        request.InputStream.Read(bytes, 0, (int)request.ContentLength64);
+                        profileDocument.content = bytes;
+                        profileDocument.contentType = request.ContentType;
+
+                        if (serverDelegate != null)
+                        {
+                            serverDelegate.AlterAgentProfileResponse(request, ref response, ref profileDocument);
+                            WriteToStream(response, contentBytes: null, ContentType.Json, HttpStatusCode.NoContent);
```

Comment:
@bbeversdorf Why not use `SendResponse` here? It looks intentional...but I can't figure it out...

Reply:
If a delegate is defined, it controls the response; it would be responsible for sending it to an actual LRS if necessary.

Reply:
Right...but `WriteToStream` and `SendResponse` would both seem to do the same thing in this case.
`SendResponse` looks like a "convenience" for sending an empty response back to the requestor (which in this case would be the webview in the mobile app).

Reply:
:+1: